common/crypto.hpp

```diff
@@ -37,6 +37,22 @@ enum CryptoErrorCode {
 	VerificationError,
 };
 
+class PrivateKey;
+using ExpectedPrivateKey = expected::expected<PrivateKey, error::Error>;
+
+class PrivateKey {
+public:
+	static ExpectedPrivateKey LoadFromPEM(const string &private_key_path, const string &passphrase);
+	static ExpectedPrivateKey LoadFromPEM(const string &private_key_path);
+#ifdef MENDER_CRYPTO_OPENSSL
+	const unique_ptr<EVP_PKEY, void (*)(EVP_PKEY *)> &pkey_;
```

Comment:
You generally don't want to use `unique_ptr` and a reference (`&`) together. While not inherently wrong, it's confusing to determine ownership, and it's usually better to use one or the other, so that you are either definitely the owner, or definitely not the owner.

`unique_ptr`s are exactly that, unique, so they only allow moving pointers into them (not copying), but that should not be a problem since your constructor takes an rvalue-reference (movable reference, in layman's terms).

Reply:
Removed the reference operator, but then I must use move in order for it to compile. Don't know why though 🤔

Reply:
Yeah, @oleorhagen and I had a long discussion about it yesterday. It's one of the dark corners of C++, and I didn't know either until we looked into it. Come talk to me if you want to know the reason! 🙂

Reply:
Yes I want to know !